New TileMap tests for 10 zoom levels, Fit Data Bounds

LeeDr opened the pull request:
Previously only 2 tests for Tile Maps.  Now 6 tests.  Additional functions in visualize_page to click zoom in and out, click the "Fit Data Bounds", and check if the zoom in and out are enabled or not.
There is one **sketchy function** to get the zoom level from the url.  It doesn't work in all cases because the uiState doesn't always contain the mapZoom parameter.

```
bdd.it('should show correct tile map data'
bdd.it('should zoom out to level 1 from default level 2'
bdd.it('"Fit data bounds" should zoom to level 3'
bdd.it('should have the correct circles on map'
bdd.it('should save with zoom level and load, take screenshot'
bdd.it('should zoom in to level 10'
```

I _could_ change visualize_page.getMapZoomLevel() to parse the query arg from the url, and get the mapZoom level from the json format instead of grepping for it.  But that wouldn't solve the issue of it not always being there.

LeeDr commented:
From the last run, minus debug logging;

```
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - should show correct tile map data (570ms)
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - should zoom out to level 1 from default level 2 (1613ms)
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - "Fit data bounds" should zoom to level 3 (1080ms)
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - should have the correct circles on map (454ms)
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - should save with zoom level and load, take screenshot (15526ms)
>> PASS: chrome on any platform - kibana - visualize app - visualize app - tile map chart - should zoom in to level 10 (5358ms)
>> 0/6 tests failed
```

epixa commented:
I don't think this mapZoom check is going to work.  With the state changing stuff that we're working on to improve IE support, it's not clear if mapZoom will even be in the url.  The concept of "current map zoom" is kind of an implementation detail of the technology rather than a feature of the product, so it's probably going to be a difficult thing to verify via integration tests.

LeeDr commented:
@epixa I removed the function for getting the zoom level from the URL and replaced it more or less with more checking of the map data.  For example in the test;

`tile map chart - should save with zoom level and load, take screenshot `

I check the data right after saving, then zoom in and check the data changed, then load the saved viz and check that we're back to the previous data.

But I'm open to other suggestions.  We could have a hidden element on the page with the zoom level that the tests could see?  Or I'm fine with what I've changed it to.
